src/components/BaseTable.vue

```diff
@@ -101,20 +102,29 @@ const pageSize = ref(kDefaultPageSize)
 const currentPage = ref(0)
 
 const sortBy = ref(props.initialSortBy)
+const sortByType = ref('string')
 const sortDirection = ref(ESortDirections.desc)
 
 const isEmpty = computed(() => props.isLoading || !props.items.length || props.isError)
 
 const sortedItems = computed(() => {
   const sortKey = sortBy.value
 
-  const itemsCopy = [...props.items]
-
-  return itemsCopy.sort((a, b) => {
+  return [...props.items].sort((a, b) => {
     if (a[sortKey] === b[sortKey]) {
       return 0
     }
 
+    if (sortByType.value === 'date') {
+      const isAfter = dayjs(a[sortKey]).isAfter(dayjs(b[sortKey]))
+
+      if (sortDirection.value === ESortDirections.desc) {
+        return isAfter ? -1 : 1
```

Comment:
I think it might be better to return a timestamp instead of `1` or `-1` 
something like this 
`a[sortKey].getTime() - b[sortKey].getTime()`